scripts/detect_duplicate_features.py

```diff
@@ -0,0 +1,73 @@
+import os
+
+import yaml
+
+
+def findall_features(features):
+    feature_list = []
+    for feature in features:
+        if "and" in feature:
+            and_list = findall_features(feature["and"])
+            for x in and_list:
+                feature_list.append(x)
+        elif "or" in feature:
+            or_list = findall_features(feature["or"])
+            for y in or_list:
+                feature_list.append(y)
+        else:
+            feature_list.append(feature)
+
+    return feature_list
+
+
+def find_overlapping_rules(new_rule_path, rules_path):
+    if not new_rule_path.endswith(".yml"):
+        return "ERROR ! New rule path file name incorrect"
+
+    count = 0
+
+    with open(new_rule_path, "r") as f:
+        new_rule = yaml.safe_load(f)
+
+    if "rule" not in new_rule:
+        return "ERROR ! given new rule path isn't a rule"
+
+    new_rule_features = findall_features(new_rule["rule"]["features"])
+
+    overlapping_rules = []
+
+    for dirpath, dirnames, filenames in os.walk(rules_path):
+        for filename in filenames:
+            if filename.endswith(".yml"):
+                rule_path = os.path.join(dirpath, filename)
+                with open(rule_path, "r") as f:
+                    rule = yaml.safe_load(f)
+                    if "rule" not in rule:
+                        continue
+                    rule_features = findall_features(rule["rule"]["features"])
+                    count += 1
+                if any([feature in rule_features for feature in new_rule_features]):
```

Comment:
I like this simple check for a first iteration of this script!

Reply:
Hi, @mr-tz 
I tried to keep it simple and focused on the main task at hand. I'm glad you found it useful. I can add more features like it will return a list of features it found similar.

Reply:
it looks good as is :)